src/basic/ether-addr-util.h

```diff
@@ -1,11 +1,31 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
 #pragma once
 
+#include <linux/if_infiniband.h>
 #include <net/ethernet.h>
 #include <stdbool.h>
 
 #include "hash-funcs.h"
 
+#define HW_ADDR_MAX_SIZE CONST_MAX(ETH_ALEN, INFINIBAND_ALEN)
```

Comment:
I noticed that linux/netdevice.h has `MAX_ADDR_LEN`. Maybe we should use it.

Reply:
That's unfortunately a bit problematic.
net/if_arp.h also defines MAX_ADDR_LEN, but as 7. So using that macro seems risky.

Reply:
Ugh... So, let's explicitly define this 32, and add a comment about why we cannot use MAX_ADDR_LEN here.